packages/build-utils/src/convert-runtime-to-plugin.ts

```diff
@@ -89,6 +94,16 @@ export function convertRuntimeToPlugin(
       // @ts-ignore This symbol is a private API
       const lambdaFiles: Files = output[FILES_SYMBOL];
 
+      // When deploying, the `files` that are passed to the Legacy Runtimes already
+      // have certain files that are ignored stripped, but locally, that list of
+      // files isn't used by the Legacy Runtimes, so we need to apply the filters
+      // to the outputs that they are returning instead.
+      for (const file in lambdaFiles) {
+        if (shouldIgnorePath(file, ignoreFilter)) {
+          delete files[file];
+        }
+      }
```

Comment:
Could it be possible, that `.vercelignore` contains some build output, that the user just doesn't want to upload, but should be part of the lambda, since it's created at build time?

Reply:
Yes, that's a good point. I think we only want to exclude the native ones then.

Done in https://github.com/vercel/vercel/pull/7114/commits/406ef8877e6e48622ddae4cd2c99643d5c8b0d18!